web/features/search/SearchContext.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  useCallback,
+} from "react";
+import { getClient } from "../../utils/sanity/sanity.server";
+
+export type SearchResultType = {
+  slug: {
+    current: string;
+  };
+  title: string;
+  availableFrom: string;
+};
+
+const SearchContext = createContext(null);
+
+type SearchContextType = {
+  searchStr: string;
+  setSearchStr: (searchStr: string) => void;
+  loading: boolean;
+  searchResults: SearchResultType[];
+  searchIsActive: boolean;
+  search: () => void;
+  closeSearch: () => void;
+};
+
+export const useSearch = (): SearchContextType => {
+  const context: SearchContextType | null = useContext(SearchContext);
+  if (context === null) {
+    throw new Error("useSearch must be used within a SearchProvider");
+  }
+  return {
+    searchStr: context.searchStr,
+    setSearchStr: context.setSearchStr,
+    loading: context.loading,
+    searchResults: context.searchResults,
+    searchIsActive: context.searchIsActive,
+    search: context.search,
+    closeSearch: context.closeSearch,
+  };
+};
+
+export const SearchProvider = ({ children }) => {
+  const [searchStr, setSearchStr] = useState<string>("");
+  const [searchResults, setSearchResults] = useState<SearchResultType[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [searchIsActive, setSearchIsActive] = useState<boolean>(false);
+
+  const search = useCallback(async () => {
+    if (searchStr.trim().length > 0) {
+      setSearchIsActive(true);
+      setLoading(true);
+      const client = getClient();
+      const results = await client.fetch<
+        SearchResultType[]
+      >(`*[[title, category, tags] match ["${searchStr}*"]]{
+        slug,
+        title,
+        availableFrom,
+      }`);
```

Comment:
Her kan du helt sikkert tweake litt mer rundt vekting av søkeresultater, slik at matcher i titler  veier mer enn tekst f.eks. . Les rundt "boost"- og "score"-funksjonene: https://www.sanity.io/docs/query-cheat-sheet

Reply:
Uansett noe vi kan tweake på etterhvert.

Reply:
Cool! Sjekker det ut

Reply:
Implementert et førsteutkast for søk med scores 👍